crates/forge/src/runner.rs

```diff
@@ -894,3 +916,50 @@ fn fuzzer_with_cases(
         TestRunner::new(config)
     }
 }
+
+/// Holds data about a persisted invariant failure.
+#[derive(Serialize, Deserialize)]
+struct InvariantPersistedFailure {
+    /// Recorded counterexample.
+    call_sequence: Vec<BaseCounterExample>,
+    /// Bytecode of the test contract that generated the counterexample.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    driver_bytecode: Option<Bytes>,
+}
+
+/// Helper function to load failed call sequence from file.
+/// Ignores failure if generated with different test contract than the current one.
+fn persisted_call_sequence(path: &Path, bytecode: &Bytes) -> Option<Vec<BaseCounterExample>> {
+    foundry_common::fs::read_json_file::<InvariantPersistedFailure>(path).ok().and_then(
+        |persisted_failure| {
+            if let Some(persisted_bytecode) = &persisted_failure.driver_bytecode {
+                // Ignore persisted sequence if test bytecode doesn't match.
+                if !bytecode.eq(persisted_bytecode) {
+                    let _= sh_warn!("\
+                            Failure from {:?} file was ignored because test contract bytecode doesn't match.",
```

Comment:
nit: `because test contract bytecode doesn't match` could be worded differently, `because test contract bytecode has changed`.

Reply:
yep, better, changed in https://github.com/foundry-rs/foundry/pull/9981/commits/520c9cee1b681e150b9333a4785ceefec972f160